resolwe_bio/processes/support_processors/seqtk_reverse_complement.py

```diff
@@ -0,0 +1,185 @@
+
+        _, _, stderr = Cmd['fastqc'][f'{name}_complemented.fastq', '--extract', '--outdir=./'] & TEE
+        if 'Failed to process' in stderr or 'Skipping' in stderr:
+            self.error('Failed while processing with FastQC.')
+
+        (Cmd['gzip'][f'{name}_complemented.fastq'])()
+
+        outputs.fastq = [f'{name}_complemented.fastq.gz']
+        outputs.fastqc_url = [f'{name}_complemented_fastqc.html']
+        outputs.fastqc_archive = [f'{name}_complemented_fastqc.zip']
+
+
+class ReverseComplementPaired(Process):
+    """Reverse complement paired-end FASTQ reads file using Seqtk."""
+
+    slug = 'seqtk-rev-complement-paired'
+    process_type = 'data:reads:fastq:paired:seqtk'
+    name = 'Reverse complement FASTQ (paired-end)'
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/common:1.3.1'
+            },
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 16384,
+        },
+    }
+    data_name = '{{ reads|sample_name|default("?") }}'
+    version = '1.0.0'
+
+
+    class Input:
+        """Input fields to process ReverseComplementPaired."""
+
+        reads = DataField('reads:fastq:paired', label='Reads')
+        select_mate = StringField(
+            label='Select mate',
+            description='Select the which mate should be reverse complemented.',
+            choices=[
+                ('Mate 1', 'Mate 1'),
+                ('Mate 2', 'Mate 2'),
+                ('Both', 'Both')
+            ],
+            default='Mate 1',
+        )
+
+    class Output:
+        """Output fields."""
+
+        fastq = ListField(FileField(), label='Reverse complemented FASTQ file')
+        fastq2 = ListField(FileField(), label='Remaining mate')
+        fastqc_url = ListField(FileHtmlField(), label='Quality control with FastQC (Mate 1)')
+        fastqc_archive = ListField(FileField(), label='Download FastQC archive (Mate 1)')
+        fastqc_url2 = ListField(FileHtmlField(), label='Quality control with FastQC (Mate 2)')
+        fastqc_archive2 = ListField(FileField(), label='Download FastQC archive (Mate 2)')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        basename_mate1 = os.path.basename(inputs.reads.fastq[0].path)
+        basename_mate2 = os.path.basename(inputs.reads.fastq2[0].path)
+        assert basename_mate1.endswith('.fastq.gz')
+        assert basename_mate2.endswith('.fastq.gz')
+        name_mate1 = basename_mate1[:-9]
+        name_mate2 = basename_mate2[:-9]
+
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq]] > f'{name_mate1}_original.fastq.gz')()
+        (Cmd['cat'][[reads.path for reads in inputs.reads.fastq2]] > f'{name_mate2}_original.fastq.gz')()
+
+        if inputs.select_mate == 'Mate 1':
+            (Cmd['seqtk']['seq', '-r', f'{name_mate1}_original.fastq.gz'] > f'{name_mate1}_complemented.fastq')()
```

Comment:
Think about adding output strings into variables and use those throughout the rest of the process.